app/controllers/application_controller.rb

```diff
@@ -1,5 +1,11 @@
 class ApplicationController < ActionController::Base
+  before_filter :set_mailer_host
+
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def set_mailer_host
+    ActionMailer::Base.default_url_options[:host] = request.host_with_port
```

Comment:
Ok. If I set it in either of those locations then it will be a literal assignment, right? Suppose that's good enough if made per env. I liked the idea of using the request to provide the host info. Seemed more flexible.